smsclient_core/smsclient.py

```diff
@@ -0,0 +1,440 @@
+        ('1', '1'),
+        ('2', '2'),
+        ('3', '3')
+        ], 'Priority', help='The priority of the message ')
+    coding = fields.Selection([
+        ('1', '7 bit'),
+        ('2', 'Unicode')
+        ], 'Coding', help='The sms coding: 1 for 7 bit or 2 for unicode')
+    tag = fields.Char('Tag', size=256,
+                      help='An optional tag')
+    nostop = fields.Boolean(
+        'NoStop',
+        help='Do not display STOP clause in the message, this requires that'
+             'this is not an advertising message')
+
+
+class HistoryLine(models.Model):
+    _name = 'sms.smsclient.history'
+    _description = 'SMS Client History'
+
+    @api.model
+    def _get_default_user(self):
+        return self.env.uid
+
+    name = fields.Char('Description', size=160, required=True, readonly=True)
+    date_create = fields.Datetime(
+        'Date',
+        readonly=True,
+        default=fields.Datetime.now)
+    user_id = fields.Many2one('res.users', 'Username', readonly=True,
+                              select=True, default=_get_default_user)
+    gateway_id = fields.Many2one(
+        'sms.smsclient',
+        'SMS Gateway',
+        help='Do not display STOP clause in the message, this requires that'
+             'this is not an advertising message')
+
+
+class Properties(models.Model):
+    _name = 'sms.smsclient.parms'
+    _description = 'SMS Client Properties'
+
+    name = fields.Char('Property name', size=256,
+                       help='Name of the property whom appear on the URL')
+    value = fields.Char('Property value', size=256,
+                        help='Value associate on the property for the URL')
+    gateway_id = fields.Many2one('sms.smsclient', 'SMS Gateway')
+    type = fields.Selection([
+        ('user', 'User'),
+        ('password', 'Password'),
+        ('sender', 'Sender Name'),
+        ('to', 'Recipient No'),
+        ('sms', 'SMS Message'),
+        ('extra', 'Extra Info')
+        ], 'API Method', select=True,
+        help='If parameter concern a value to substitute, indicate it')
+
+
+class HistoryLine(models.Model):
+    _name = 'sms.smsclient.history'
+    _description = 'SMS Client History'
+
+    @api.model
+    def _get_default_user(self):
+        return self.env.uid
+
+    name = fields.Char('Description', size=160, required=True, readonly=True)
+    date_create = fields.Datetime(
+        'Date',
+        readonly=True,
+        default=fields.Datetime.now)
+    user_id = fields.Many2one('res.users', 'Username', readonly=True,
+                              select=True, default=_get_default_user)
+    gateway_id = fields.Many2one(
+        'sms.smsclient',
+        'SMS Gateway',
+        ondelete='set null',
+        required=True)
+    to = fields.Char('Mobile No', size=15, readonly=True)
+    sms = fields.Text('SMS', size=160, readonly=True)
+
+    @api.model
+    def create(self, vals):
+        super(HistoryLine, self).create(vals)
+        self.env.cr.commit()
```

Comment:
You should certainly remove this commit and return the result of super. Or is there any reason to do it this way? If so a comment would be very appreciated but I think this is strange.